pkg/doc/signature/proof/jws.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package proof
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"strings"
+
+	"github.com/piprate/json-gold/ld"
+)
+
+const securityContext = "https://w3id.org/security/v2"
+
+// CreateDetachedJWTHeader creates detached JWT header.
+func CreateDetachedJWTHeader(p *Proof) string {
+	jwtHeaderMap := map[string]interface{}{
+		"alg":  p.Type,
+		"b64":  false,
+		"crit": []string{"b64"},
+	}
+
+	jwtHeaderBytes, _ := json.Marshal(jwtHeaderMap) //nolint:errcheck
+
+	return base64.RawURLEncoding.EncodeToString(jwtHeaderBytes)
+}
+
+// GetJWTSignature returns signature part of JWT
+func GetJWTSignature(jwt string) ([]byte, error) {
+	jwtParts := strings.Split(jwt, ".")
+	if len(jwtParts) != 3 || jwtParts[2] == "" {
+		return nil, errors.New("invalid JWT")
+	}
+
+	return base64.RawURLEncoding.DecodeString(jwtParts[2])
+}
+
+func getJWTHeader(jwt string) (string, error) {
+	jwtParts := strings.Split(jwt, ".")
+	if len(jwtParts) != 3 { // nolint:gomnd
+		return "", errors.New("invalid JWT")
+	}
+
+	return jwtParts[0], nil
```

Comment:
also create constants for token parts such as 0 should be jwtHeaderPart, 2 should be jwtSignaturePart etc.

Reply:
fixed